### ISSUE TITLE
The last 25% of the test data set is all predicted to be wrong

### ISSUE THREAD

**Issue by dunfrey** [keras-team/keras#13077]
I trained a CNN and everything occurs well, except the loss validation of one kind of output (training loss decreases but the validation loss not decrease, is too high and almost constant). I'd tried everything: more dropouts, decrease learning rate, increase and decrease the complexity of the nn, regularizers...  

Nothing solves, so I just wait for the end of the training step. On the test, is everything predicted close to the correct label but the lasts are every predicted wrong.  

EXAMPLE:
```
9790/9790 [==============================] - 33s 3ms/step - loss: 104.9522 - acc: 0.8920 - val_loss: 13.6372 - val_acc: 0.8994
Epoch 2/5
9790/9790 [==============================] - 24s 2ms/step - loss: 2.2317 - acc: 0.9354 - val_loss: 16.8571 - val_acc: 0.8904
Epoch 3/5
9790/9790 [==============================] - 24s 2ms/step - loss: 0.2924 - acc: 0.9830 - val_loss: 14.4873 - val_acc: 0.8696
Epoch 4/5
9790/9790 [==============================] - 24s 2ms/step - loss: 0.1365 - acc: 0.9884 - val_loss: 13.8804 - val_acc: 0.8735
Epoch 5/5
9790/9790 [==============================] - 24s 2ms/step - loss: 0.0587 - acc: 0.9896 - val_loss: 14.1104 - val_acc: 0.8775
```

For example, I have 100 images. The first 75 images are predicted close to the ground truth, but the last 25% data is predicted really far from the correct label (100 images - last 25%). So, I remove this last 25% of data from the test, and now, of the new set data to test (75 images), the last 25% is predicted wrong.

When I set to 100 epochs, the results look like this:
```
Epoch 5/100
9790/9790 [==============================] - 24s 2ms/step - loss: 0.0587 - acc: 0.9896 - val_loss: 14.1104 - val_acc: 0.8775
[...]
Epoch 100/100
9790/9790 [==============================] - 24s 2ms/step - loss: 0.0019 - acc: 0.9915 - val_loss: 15.2133 - val_acc: 0.8775

```
Sometimes the validation loss changes but not so far that the result that was shown and is very far to the training validation.

**Comment by dabasajay** [comment]
If possible, can you provide the source code? Also, did you try to shuffle the dataset?

**Comment by dunfrey** [comment]
Yes. The network is PoseNet, that is basically a GoogLeNet without the intermediate outputs and with two outputs on the end. The objective is use a image as input and predict the position of the camera (it is the label). Attahed the NN:

```
with tf.device('/device:GPU:0'):
        inputs = Input((IMG_HEIGHT, IMG_WIDTH, IMG_CHANNELS))
        init     = Conv2D(64,(7,7),padding='same',activation='relu',name='init/conv1')(inputs)
        init     = MaxPooling2D(pool_size=(3,3),strides=(2,2),padding='valid',name='init/max1')(init)
        init     = BatchNormalization(name='init/norm1')(init)    
        init     = Conv2D(64, (1,1),padding='same',activation='relu',name='init/conv2_1')(init)
        init     = Conv2D(192,(3,3),padding='same',activation='relu',name='init/conv2_3')(init)
        init     = BatchNormalization(name='init/nomr2')(init)
        init_out = MaxPooling2D(pool_size=(3,3),strides=(2,2),padding='valid',name='init/3x3')(init)
        
        #inception 1
        icp_1_1x1        = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_1/1x1')(init_out)
        icp_1_3x3_reduce = Conv2D(96, (1,1),padding='same',activation='relu',name='icp_1/3x3_reduce')(init_out)
        icp_1_3x3        = Conv2D(128,(3,3),padding='same',activation='relu',name='icp_1/3x3')(icp_1_3x3_reduce)
        icp_1_5x5_reduce = Conv2D(16, (1,1),padding='same',activation='relu',name='icp_1/5x5_reduce')(init_out)
        icp_1_5x5        = Conv2D(32, (5,5),padding='same',activation='relu',name='icp_1/5x5')(icp_1_3x3_reduce)
        icp_1_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_1/pool')(init_out)
        icp_1_pool_proj  = Conv2D(32, (1,1),padding='same',activation='relu',name='icp_1/pool_proj')(icp_1_pool)
        icp_1_output     = concatenate([icp_1_1x1,icp_1_3x3,icp_1_5x5,icp_1_pool_proj],name='icp_1/output')
        
        #inception 2
        icp_2_1x1        = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_2/1x1')(icp_1_output)
        icp_2_3x3_reduce = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_2/3x3_reduce')(icp_1_output)
        icp_2_3x3        = Conv2D(192,(3,3),padding='same',activation='relu',name='icp_2/3x3')(icp_2_3x3_reduce)
        icp_2_5x5_reduce = Conv2D(32, (1,1),padding='same',activation='relu',name='icp_2/5x5_reduce')(icp_1_output)
        icp_2_5x5        = Conv2D(96, (5,5),padding='same',activation='relu',name='icp_2/5x5')(icp_2_5x5_reduce)
        icp_2_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_2/pool')(icp_1_output)
        icp_2_pool_proj  = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_2/pool_proj')(icp_2_pool)
        icp_2_output     = concatenate([icp_2_1x1,icp_2_3x3,icp_2_5x5,icp_2_pool_proj],name='icp_2/output')
        
        pool2_3x3_icp3   = MaxPooling2D(pool_size=(3,3),strides=(2,2),padding='valid',name='pool3/3x3_s2')(icp_2_output)
        
        #inception 3
        icp_3_1x1        = Conv2D(192,(1,1),padding='same',activation='relu',name='icp_3/1x1')(pool2_3x3_icp3)
        icp_3_3x3_reduce = Conv2D(96, (1,1),padding='same',activation='relu',name='icp_3/3x3_reduce')(pool2_3x3_icp3)
        icp_3_3x3        = Conv2D(208,(3,3),padding='same',activation='relu',name='icp_3/3x3')(icp_3_3x3_reduce)
        icp_3_5x5_reduce = Conv2D(16, (1,1),padding='same',activation='relu',name='icp_3/5x5_reduce')(pool2_3x3_icp3)
        icp_3_5x5        = Conv2D(48, (5,5),padding='same',activation='relu',name='icp_3/5x5')(icp_3_5x5_reduce)
        icp_3_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_3/pool')(pool2_3x3_icp3)
        icp_3_pool_proj  = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_3/pool_proj')(icp_3_pool)
        icp_3_output     = concatenate([icp_3_1x1,icp_3_3x3,icp_3_5x5,icp_3_pool_proj],name='icp_3/output')

        #inception 4
        icp_4_1x1        = Conv2D(160,(1,1),padding='same',activation='relu',name='icp_4/1x1')(icp_3_output)
        icp_4_3x3_reduce = Conv2D(112,(1,1),padding='same',activation='relu',name='icp_4/3x3_reduce')(icp_3_output)
        icp_4_3x3        = Conv2D(224,(3,3),padding='same',activation='relu',name='icp_4/3x3')(icp_4_3x3_reduce)
        icp_4_5x5_reduce = Conv2D(24, (1,1),padding='same',activation='relu',name='icp_4/5x5_reduce')(icp_3_output)
        icp_4_5x5        = Conv2D(64, (5,5),padding='same',activation='relu',name='icp_4/5x5')(icp_4_5x5_reduce)
        icp_4_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_4/pool')(icp_3_output)
        icp_4_pool_proj  = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_4/pool_proj')(icp_4_pool)
        icp_4_output     = concatenate([icp_4_1x1,icp_4_3x3,icp_4_5x5,icp_4_pool_proj],name='icp_4_output')
        
        #inception 5
        icp_5_1x1        = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_5/1x1')(icp_4_output)
        icp_5_3x3_reduce = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_5/3x3_reduce')(icp_4_output)
        icp_5_3x3        = Conv2D(256,(3,3),padding='same',activation='relu',name='icp_5/3x3')(icp_5_3x3_reduce)
        icp_5_5x5_reduce = Conv2D(24, (1,1),padding='same',activation='relu',name='icp_5/5x5_reduce')(icp_4_output)
        icp_5_5x5        = Conv2D(64, (5,5),padding='same',activation='relu',name='icp_5/5x5')(icp_5_5x5_reduce)
        icp_5_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_5/pool')(icp_4_output)
        icp_5_pool_proj  = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_5/pool_proj')(icp_5_pool)
        icp_5_output     = concatenate([icp_5_1x1,icp_5_3x3,icp_5_5x5,icp_5_pool_proj],name='icp_5/output')
        
        #inception 6
        icp_6_1x1        = Conv2D(112,(1,1),padding='same',activation='relu',name='icp_6/1x1')(icp_5_output)
        icp_6_3x3_reduce = Conv2D(144,(1,1),padding='same',activation='relu',name='icp_6/3x3_reduce')(icp_5_output)
        icp_6_3x3        = Conv2D(288,(3,3),padding='same',activation='relu',name='icp_6/3x3')(icp_6_3x3_reduce)
        icp_6_5x5_reduce = Conv2D(32, (1,1),padding='same',activation='relu',name='icp_6/5x5_reduce')(icp_5_output)
        icp_6_5x5        = Conv2D(64, (5,5),padding='same',activation='relu',name='icp_6/5x5')(icp_6_5x5_reduce)
        icp_6_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_6/pool')(icp_5_output)
        icp_6_pool_proj  = Conv2D(64, (1,1),padding='same',activation='relu',name='icp_6/pool_proj')(icp_6_pool)
        icp_6_output     = concatenate([icp_6_1x1,icp_6_3x3,icp_6_5x5,icp_6_pool_proj],name='icp_6/output')

        #inception 7
        icp_7_1x1        = Conv2D(256,(1,1),padding='same',activation='relu',name='icp_7/1x1')(icp_6_output)
        icp_7_3x3_reduce = Conv2D(160,(1,1),padding='same',activation='relu',name='icp_7/3x3_reduce')(icp_6_output)
        icp_7_3x3        = Conv2D(320,(3,3),padding='same',activation='relu',name='icp_7/3x3')(icp_7_3x3_reduce)
        icp_7_5x5_reduce = Conv2D(32, (1,1),padding='same',activation='relu',name='icp_7/5x5_reduce')(icp_6_output)
        icp_7_5x5        = Conv2D(128,(5,5),padding='same',activation='relu',name='icp_7/5x5')(icp_7_5x5_reduce)
        icp_7_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_7/pool')(icp_6_output)
        icp_7_pool_proj  = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_7/pool_proj')(icp_7_pool)
        icp_7_output     = concatenate([icp_7_1x1,icp_7_3x3,icp_7_5x5,icp_7_pool_proj],name='icp_7/output')

        pool3_3x3_icp8   = MaxPooling2D(pool_size=(3,3),strides=(2,2),padding='valid',name='pool4/3x3_s2')(icp_7_output)
        
        #inception 8
        icp_8_1x1        = Conv2D(256,(1,1),padding='same',activation='relu',name='icp_8/1x1')(pool3_3x3_icp8)
        icp_8_3x3_reduce = Conv2D(160,(1,1),padding='same',activation='relu',name='icp_8/3x3_reduce')(pool3_3x3_icp8)
        icp_8_3x3        = Conv2D(320,(3,3),padding='same',activation='relu',name='icp_8/3x3')(icp_8_3x3_reduce)
        icp_8_5x5_reduce = Conv2D(32, (1,1),padding='same',activation='relu',name='icp_8/5x5_reduce')(pool3_3x3_icp8)
        icp_8_5x5        = Conv2D(128,(5,5),padding='same',activation='relu',name='icp_8/5x5')(icp_8_5x5_reduce)
        icp_8_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_8/pool')(pool3_3x3_icp8)
        icp_8_pool_proj  = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_8/pool_proj')(icp_8_pool)
        icp_8_output     = concatenate([icp_8_1x1,icp_8_3x3,icp_8_5x5,icp_8_pool_proj],name='icp_8/output')
        
        #inception 9
        icp_9_1x1        = Conv2D(384,(1,1),padding='same',activation='relu',name='icp_9/1x1')(icp_8_output)
        icp_9_3x3_reduce = Conv2D(192,(1,1),padding='same',activation='relu',name='icp_9/3x3_reduce')(icp_8_output)
        icp_9_3x3        = Conv2D(384,(3,3),padding='same',activation='relu',name='icp_9/3x3')(icp_9_3x3_reduce)
        icp_9_5x5_reduce = Conv2D(48, (1,1),padding='same',activation='relu',name='icp_9/5x5_reduce')(icp_8_output)
        icp_9_5x5        = Conv2D(128,(5,5),padding='same',activation='relu',name='icp_9/5x5')(icp_9_5x5_reduce)
        icp_9_pool       = MaxPooling2D(pool_size=(3,3),strides=(1,1),padding='same',name='icp_9/pool')(icp_8_output)
        icp_9_pool_proj  = Conv2D(128,(1,1),padding='same',activation='relu',name='icp_9/pool_proj')(icp_9_pool)
        icp_9_output     = concatenate([icp_9_1x1,icp_9_3x3,icp_9_5x5,icp_9_pool_proj],name='icp_9/output')
        
        cls_pose = AveragePooling2D(pool_size=(7,7),strides=(1,1),name='pool5/7x7_s2')(icp_9_output)
        cls_pose = Flatten()(cls_pose)
        cls_pose = Dropout(50)(cls_pose)
        
        cls_pose_xyz  = Dense(2,name='cls3_fc_pose_xyz')(cls_pose)
        cls_pose_wpqr = Dense(2,name='cls3_fc_pose_wpqr')(cls_pose)

        model_posenet = Model(inputs=[inputs], outputs=[cls_pose_xyz, cls_pose_wpqr])

        adam = optimizers.Adam(lr = 1e-3, amsgrad = True)  
        model_posenet.compile(optimizer = adam, 
                              loss = ['mse', 
                                      'mse'],
                              metrics = ['accuracy'])
```
The problem occours with the output **cls_pose_xyz  = Dense(2,name='cls3_fc_pose_xyz')(cls_pose)**
This values espected have the range of:
[[ -4.26460441668e-05 , 10.4308511841 ] , [ -9.26670750683e-06 , 3.92646981481 ]]

To train I'm using:

```
results_posenet = model_posenet.fit(X_train, 
                      [y_train_axis_pose, y_train_quat], 
                      validation_split = .3, 
                      batch_size       = 64, 
                      epochs           = 100, 
                      suffle           = True,
                      callbacks        = [checkpointer_pn])
```

To image preprocesses I just resizing the image:
```
def prepare_image(images_):
    image = np.zeros((len(images_), IMG_HEIGHT, IMG_WIDTH, IMG_CHANNELS), dtype=np.uint8)
    for i in tqdm(range(len(images_))):
        X = cv2.imread(images_[i])
        X = cv2.resize(X, (IMG_HEIGHT, IMG_WIDTH))
        image[i] = (X)
    image_ = image.astype('float32')
    image_ /= 255
    return image_
```

The accuracy in training is changing as weel the loss too, but in validation the accuracy is the same every time and the validation changes but is too high yet:
**TRAINING:**
`cls3_fc_pose_xyz_loss: 0.1060 - cls3_fc_pose_wpqr_loss: 0.0388 - cls3_fc_pose_xyz_accuracy: 0.9826 - cls3_fc_pose_wpqr_accuracy: 0.9487`
**VALIDATION:**
`val_cls3_fc_pose_xyz_loss: 15.2133 - val_cls3_fc_pose_wpqr_loss: 0.5394 - val_cls3_fc_pose_xyz_accuracy: 0.8775 - val_cls3_fc_pose_wpqr_accuracy: 0.6888`
